ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java

```diff
@@ -194,6 +197,22 @@ void compact(CompactionDetails details, byte[] bundleReadBuffer) throws IOExcept
     resumeCompaction(bundleReadBuffer);
   }
 
+  /**
+   * Filters not active {@link Container}s for compaction purpose. Not active status include DELETE_IN_PROGRESS and INACTIVE.
+   * @return the not active {@link Container}s' accountId & containerId pairs.
+   */
+  private void getNotActiveContainers() {
+    notActiveContainers.clear();
+    if (accountService != null) {
+      accountService.getContainersByStatus(Container.ContainerStatus.DELETE_IN_PROGRESS).forEach((container) -> {
```

Comment:
Sorry, two additional minor questions:
1. `accountService` is not initialized in this PR, is it?
2. I think HelixAccountService may fail to fetch account metadata (i.e.disconnected from ZK), I think the accountService will get containers from local backup file, right?

Reply:
1. No, it's in previous PR initialized inside Ambry Server.
2. Yes, the account metadata will be cached locally, so serving an Account query will not incur any calls to remote Zookeeper.